### PR TITLE
add column for std nomenclature

### PR DESCRIPTION
**Pull request by tedsharpe**

### CODE REVIEW

**Comment by tedsharpe** [comment]
fixed unit test

File: src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java [hunk 1]
```diff
@@ -934,6 +978,112 @@ public static CodonVariation createModification( final int codonId, final int co
...
+            startingCodon = endingCodon = codonVariation.getCodonId();
+        }
+
+        // attempt to add more variations
+        // returns false if the presented variation cannot be added to the group
+        public boolean addVariation( final CodonVariation codonVariation ) {
+            final int codonId = codonVariation.getCodonId();
+            if ( codonId > endingCodon + 1 && !isFrameShift ) return false;
+            switch ( codonVariation.getVariationType() ) {
+                case FRAMESHIFT:
```
Comment (SHuang-Broad) [inline src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java]: I am confused by the asymmetry between the above ctor and this modifier.
The ctor allows creation of a group with codon variation that is a `FRAMESHIFT`.
But here regardless if the group was constructed by a `FRAMESHIFT`, it always returns false.
I must be not understanding something.
Can you elaborate?

File: src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java [hunk 2]
```diff
@@ -934,6 +978,112 @@ public static CodonVariation createModification( final int codonId, final int co
...
+            }
+            endingCodon = codonId;
+            return true;
+        }
+
+        // sometimes when starting with a frame shift you end up with nothing because the variants are all synonymous
+        public boolean isEmpty() { return subCount + insCount + delCount == 0; }
+
+        // translate the group into standard nomenclature
+        @Override public String toString() {
```
Comment (SHuang-Broad) [inline src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java]: may I suggest a different name for this method?
`tostring()` to me is almost always for debugging.
This method is clearly doing very important tasks.

What about `toStandardNomanclature()`?

File: src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java [hunk 3]
```diff
@@ -103,6 +103,9 @@
     @Argument(doc = "minimum number of wt calls flanking variant", fullName = "min-flanking-length")
     @VisibleForTesting static int minFlankingLength = 2;
 
+    @Argument(doc = "minimum map quality for read alignment", fullName = "min-mapq")
```
[the remark this answers is not in the document]
Comment (tedsharpe) [reply src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java]: i'll try to amplify a little, but the word "inclusive" used here and not, for example, for the minFlankingLength arg just 3 lines above, will just confuse people, i fear.  i think that "must be greater than or equal" is what minimum means, right?

File: src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java [hunk 4]
```diff
@@ -335,6 +345,36 @@ private static void describeVariantsAsCodons( final BufferedWriter writer, final
...
+                }
+            }
+        }
+        if ( codonVariationGroup != null && !codonVariationGroup.isEmpty() ) {
+            writer.write(sep);
+            writer.write(codonVariationGroup.toString());
+        }
+    }
+
+    private static boolean isSynonymous( final CodonVariation variation, final int[] refCodonValues ) {
```
[the remark this answers is not in the document]
Comment (tedsharpe) [reply src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java]: SNVs in coding regions are either synonymous (they don't change the amino acid incorporated into the protein), missense (the base change changes the AA), or nonsense (the base change creates a stop code that truncates the protein).
I'll add a comment that includes the words "silent mutation".

File: src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java [hunk 5]
```diff
@@ -934,6 +978,112 @@ public static CodonVariation createModification( final int codonId, final int co
...
+        private boolean isFrameShift;
+        private int insCount;
+        private int delCount;
+        private int subCount;
+
+        // start a new group with a first variation
+        public CodonVariationGroup( final int[] refCodonValues, final CodonVariation codonVariation ) {
+            this.refCodonValues = refCodonValues;
+            altCalls = new StringBuilder();
+            switch ( codonVariation.getVariationType() ) {
```
[the remark this answers is not in the document]
Comment (tedsharpe) [reply src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java]: Sure.  That would be clearer.

File: src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java (hunk 1 above)
Comment (tedsharpe) [reply src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java]: you can never add a FRAMESHIFT to a group of variations that is being built.  a FRAMESHIFT always causes a new group to be started.  (the calling code is looping over variants, asking whether or not they can be incorporated into a growing group of variants that can be described by a single bit of standard nomenclature.  FRAMESHIFTs always stand alone.)

Comment (tedsharpe) [reply src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java]: And FRAMESHIFTs are unique.  you can only ever have one.

File: src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (SHuang-Broad) [reply src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java]: agree.

File: src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java (hunk 1 above)
Comment (SHuang-Broad) [reply src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java]: I see. Thanks for explaining.

**Pull request merged**